src/Microsoft.AspNetCore.DataProtection/KeyManagement/XmlKeyManager.cs

```diff
@@ -454,5 +451,65 @@ void IInternalXmlKeyManager.RevokeSingleKey(Guid keyId, DateTimeOffset revocatio
             KeyRepository.StoreElement(revocationElement, friendlyName);
             TriggerAndResetCacheExpirationToken();
         }
+
+        private sealed class AggregateKeyEscrowSink : IKeyEscrowSink
+        {
+            private readonly IEnumerable<IKeyEscrowSink> _sinks;
+
+            public AggregateKeyEscrowSink(IEnumerable<IKeyEscrowSink> sinks)
+            {
+                _sinks = sinks;
+            }
+
+            public void Store(Guid keyId, XElement element)
+            {
+                foreach (var sink in _sinks)
+                {
+                    sink.Store(keyId, element);
+                }
+            }
+        }
+
+        /// <summary>
+        /// A simplified default implementation of <see cref="IActivator"/> that understands
+        /// how to call ctors which take <see cref="ILoggerFactory"/>.
+        /// </summary>
+        private sealed class SimpleActivator : IActivator
```

Comment:
This needs to be removed. Needs some discussion on how to approach this scenario.